### PR TITLE
Order mark as paid ajax api created

### PR DESCRIPTION
**Pull request by mi-prakash**
image

Task Url
https://ollyo.atlassian.net/browse/TUTOR-2269

### CODE REVIEW

File: ecommerce/OrderController.php [hunk 1]
```diff
@@ -156,6 +163,48 @@ public function get_order_by_id() {
...
+	 * This function verifies a nonce for security, constructs a payload object with
+	 * the order ID, note, and payment status, and updates the payment status of the order
+	 * to 'paid'. It sends a JSON response indicating the success or failure of the operation.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return void
+	 */
+	public function order_mark_as_paid() {
+		if ( ! tutor_utils()->is_nonce_verified() ) {
```
Comment (shewa12) [inline ecommerce/OrderController.php:178]: Please Add permission check after nonce validation like below:
`
if ( ! current_user_can( 'manage_options' ) ) {
    $this->json_response( MESSAGE, null, STATUS_CODE );
}
`

File: ecommerce/OrderController.php [hunk 2]
```diff
@@ -156,6 +163,48 @@ public function get_order_by_id() {
...
+			$this->json_response( tutor_utils()->error_message( 'nonce' ), null, HttpHelper::STATUS_BAD_REQUEST );
+		}
+
+		$payload                 = new \stdClass();
+		$payload->order_id       = Input::post( 'order_id' );
+		$payload->note           = Input::post( 'note' );
+		$payload->payment_status = $this->model::PAYMENT_STATUS_PAID;
+
+		if ( empty( $payload->order_id ) ) {
```
Comment (shewa12) [inline ecommerce/OrderController.php:187]: Use validation rules & validation helper for validating inputs. Check `CourseController::validate` method from `tutor-pro` : **app/public/wp-content/plugins/tutor-pro/rest-api/Controllers/CourseController.php**

File: models/OrderModel.php [hunk 3]
```diff
@@ -519,4 +519,30 @@ public function get_order_count( $where = array(), string $search_term = '' ) {
...
+	 * @since 3.0.0
+	 *
+	 * @param object $data An object containing the payment status, note, and order ID.
+	 *                     - 'payment_status' (string): The new payment status.
+	 *                     - 'note' (string): A note regarding the payment status update.
+	 *                     - 'order_id' (int): The ID of the order to update.
+	 *
+	 * @return bool True on successful update, false on failure.
+	 */
+	public function payment_status_update( $data ) {
```
Comment (shewa12) [inline models/OrderModel.php:538]: please use type hint:
`object $data`

**Review by shewa12 — CHANGES_REQUESTED** [review]
Please check and requested changes.

File: ecommerce/OrderController.php [hunk 4]
```diff
@@ -156,6 +164,73 @@ public function get_order_by_id() {
...
+		}
+
+		do_action( 'tutor_before_order_mark_as_paid' );
+
+		$inputs = array(
+			'order_id' => Input::post( 'order_id' ),
+			'note'     => Input::post( 'note' ),
+		);
+
+		$params = Input::sanitize_array( $inputs );
```
Comment (shewa12) [inline ecommerce/OrderController.php:194]: Update this code everywhere as discussed

File: ecommerce/OrderController.php [hunk 5]
```diff
@@ -156,6 +164,73 @@ public function get_order_by_id() {
...
+			$this->json_response(
+				__( 'Order Id is required', 'tutor' ),
+				null,
+				HttpHelper::STATUS_BAD_REQUEST
+			);
+		}
+
+		$response = $this->model->payment_status_update( $payload );
+
+		do_action( 'tutor_after_order_mark_as_paid' );
```
Comment (shewa12) [inline ecommerce/OrderController.php:221]: Pass order argument

File: models/OrderModel.php [hunk 6]
```diff
@@ -519,4 +519,30 @@ public function get_order_count( $where = array(), string $search_term = '' ) {
...
+	 * @return bool True on successful update, false on failure.
+	 */
+	public function payment_status_update( object $data ) {
+		return QueryHelper::update(
+			$this->table_name,
+			array(
+				'payment_status' => $data->payment_status,
+				'note'           => $data->note,
+			),
+			array( 'id' => $data->order_id )
```
Comment (shewa12) [inline models/OrderModel.php:545]: Store order activity after each action on order

**Review by shewa12 — CHANGES_REQUESTED** [review]
Please resolve the changes

File: ecommerce/OrderController.php [hunk 7]
```diff
@@ -156,6 +164,71 @@ public function get_order_by_id() {
...
+		do_action( 'tutor_before_order_mark_as_paid', $params );
+
+		$payload                 = new \stdClass();
+		$payload->order_id       = $params['order_id'];
+		$payload->note           = $params['note'];
+		$payload->payment_status = $this->model::PAYMENT_PAID;
+
+		if ( empty( $payload->order_id ) ) {
+			$this->json_response(
+				__( 'Order Id is required', 'tutor' ),
```
Comment (shewa12) [inline ecommerce/OrderController.php:211]: `__( 'Order Id is required', 'tutor' ),` I think it would be ID or id?

**Review by shewa12 — CHANGES_REQUESTED** [review]
Please check the toast messages

File: ecommerce/OrderController.php (hunk 7 above)
Comment (mi-prakash) [reply ecommerce/OrderController.php:211]: im using ID then

**Review by shewa12 — APPROVED** [review]

**Pull request merged**